gstd/src/msg.rs

```diff
@@ -17,155 +17,87 @@
 // along with this program. If not, see <https://www.gnu.org/licenses/>.
 
 use crate::prelude::Vec;
-use crate::MessageHandle;
 use crate::{MessageId, ProgramId};
+use codec::{Decode, Encode, Output};
 
-mod sys {
-    extern "C" {
-        pub fn gr_gas_available() -> u64;
-        pub fn gr_msg_id(val: *mut u8);
-        pub fn gr_read(at: u32, len: u32, dest: *mut u8);
-        pub fn gr_reply(data_ptr: *const u8, data_len: u32, gas_limit: u64, value_ptr: *const u8);
-        pub fn gr_reply_push(data_ptr: *const u8, data_len: u32);
-        pub fn gr_reply_to(dest: *mut u8);
-        pub fn gr_send(
-            program: *const u8,
-            data_ptr: *const u8,
-            data_len: u32,
-            gas_limit: u64,
-            value_ptr: *const u8,
-            message_id_ptr: *mut u8,
-        );
-        pub fn gr_send_commit(
-            handle: u32,
-            message_id_ptr: *mut u8,
-            program: *const u8,
-
-            gas_limit: u64,
-            value_ptr: *const u8,
-        );
-        pub fn gr_send_init() -> u32;
-        pub fn gr_send_push(handle: u32, data_ptr: *const u8, data_len: u32);
-        pub fn gr_size() -> u32;
-        pub fn gr_source(program: *mut u8);
-        pub fn gr_value(val: *mut u8);
-        pub fn gr_wait() -> !;
-        pub fn gr_wake(waker_id_ptr: *const u8);
+pub use gcore::msg::{gas_available, id, reply_to, source, value, wait, wake};
```

Comment:
The idiomatic way to call functions is `module::func()`, so we don't need to import functions themselves.

Reply:
They are imported separately to be accessible as `gstd::msg::func`, this line is a re-export.

Reply:
Why not just `pub use gcore::msg;`?

Reply:
> Why not just `pub use gcore::msg;`?

Ah, got it. Ok.